### PR TITLE
Apply valid_range in MiRS reader when present

### PR DESCRIPTION
**Pull request by joleenf**
This fixes a bug in which the valid range was not used.  Fix checks for a valid range in the attributes and applies the range if present.  There is a new check in the test code which determines if valid_range is removed from the attributes after loading the data.

### CODE REVIEW

File: satpy/readers/mirs.py [hunk 1]
```diff
@@ -356,6 +356,17 @@ def _fill_data(self, data_arr, attrs):
             data_arr = data_arr.where(data_arr != fill_value, fill_out)
         return data_arr, attrs
 
+    def _get_valid_range(self, data_arr, attrs):
```
Comment (djhoese) [inline satpy/readers/mirs.py:359]:
```suggestion
    def _apply_valid_range(self, data_arr, attrs):
```

Really nitpicking here. Not necessary unless you make other changes.

File: satpy/readers/mirs.py [hunk 2]
```diff
@@ -356,6 +356,17 @@ def _fill_data(self, data_arr, attrs):
...
+        # handle valid_range
+        valid_range = attrs.pop('valid_range', None)
+        if isinstance(valid_range, np.ndarray):
```
Comment (djhoese) [inline satpy/readers/mirs.py:362]: What are the other options for what this could be? Would this catch more cases if it was `valid_range is not None`?

**Review by djhoese — COMMENTED** [review]
I had a couple questions. Also, how hard would it be to check the data in the tests to make sure the valid range was applied?

File: satpy/readers/mirs.py (hunk 2 above)
Comment (joleenf) [reply satpy/readers/mirs.py:362]: valid_range is not None is better.

**Review by djhoese — APPROVED** [review]
LGTM

**Pull request merged**